backend/proxy-server/src/server.js

```diff
@@ -44,65 +48,83 @@ app.all('/*', upload.none(), (req, res, next) => {
+    }
+    if (!allowedIP) {
+      return next(
+        new RestApiError(403, {
+          client: `${clientIp} restricted`,
+        })
+      );
     }
 
-    if (!allowed && inactive !== allowlist.length) {
-      // console.log('---->', clientIp, allowlist, inactive);
-      return next(new RestApiError(403, { client: `${clientIp} restricted` }));
+    if (allowedIP.status !== 'active') {
+      return next(
+        new RestApiError(403, {
+          client: `${clientIp} restricted`,
+        })
+      );
     }
   }
 
-  // PUT, POST and PATCH operations have records in body
-  if (['PUT', 'PATCH', 'POST'].includes(req.method) && !req.body.records) {
-    return next(new RestApiError(422, { records: 'not present' }));
+  // check racm for allowed methods
+  if (conduit.racm.findIndex((method) => method === req.method) === -1) {
+    return next(
+      new RestApiError(405, {
+        [req.method]: 'not permitted',
+      })
+    );
   }
 
-  // PUT, POST and PATCH operations need fields data in body
-  if (
-    ['PUT', 'PATCH', 'POST'].includes(req.method) &&
-    req.body.records[0].fields === undefined
-  ) {
-    return next(new RestApiError(422, { fields: 'not present' }));
+  if (['PUT', 'PATCH', 'POST'].includes(req.method)) {
+    // PUT, POST and PATCH operations have records in body
+    if (!req.body.records) {
+      return next(
+        new RestApiError(422, {
+          records: 'not present',
+        })
+      );
+    }
+
+    if (req.body.records[0].fields === undefined) {
+      // PUT, POST and PATCH operations need fields data in body
+      return next(
+        new RestApiError(422, {
+          fields: 'not present',
+        })
+      );
+    }
   }
 
   // PUT and PATCH operations need id field in body
   if (
     ['PUT', 'PATCH'].includes(req.method) &&
     req.body.records[0].id === undefined
   ) {
-    return next(new RestApiError(422, { id: 'not provided' }));
-  }
-
-  // check racm for allowed methods
-  if (conduit.racm.findIndex((method) => method === req.method) === -1) {
-    return next(new RestApiError(405, { [req.method]: 'not permitted' }));
+    return next(
+      new RestApiError(422, {
+        id: 'not provided',
+      })
+    );
   }
 
   // perform hidden form field validation (needed only for new record creation)
+
   if (req.method === 'POST') {
     for (let i = 0, imax = conduit.hiddenFormField.length; i < imax; i++) {
       // We`ll be using this multiple times, so store in a short variable
       const hff = conduit.hiddenFormField[i];
-      let reqHff = undefined;
-      if (req.body?.records?.[0].fields[hff.fieldName]) {
-        reqHff = req.body.records[0].fields[hff.fieldName];
-      }
+      const reqHff = req.body.records?.[0].fields[hff.fieldName];
 
       // This feature is to catch spam bots, so don't
       // send error if failure, send 200-OK instead
-      if (hff?.policy === 'drop-if-filled' && reqHff) {
+      if (hff.policy === 'drop-if-filled' && reqHff) {
```

Comment:
hff is already checked as the for loop executes when the hiddenFormField is present. So this part neednt be chained.